ide/languages.hcl/src/org/netbeans/modules/languages/hcl/grammar/g4/HCLExpressionParser.g4

```diff
@@ -128,13 +128,19 @@ heredoc
     : HEREDOC_START heredocTemplate HEREDOC_END
     ;
 
+// This is not part of the HCL spec, though used in Terraform 1.8 for function calls
+scopedId
+    : target=IDENTIFIER SCOPE ref=IDENTIFIER
+    | scopedId SCOPE ref=IDENTIFIER
+    ;
+
 variableExpr
     : IDENTIFIER
     ;
 
 functionCall
-    : IDENTIFIER LPAREN arguments RPAREN
-    | IDENTIFIER LPAREN RPAREN
+    : (IDENTIFIER|scopedId) LPAREN arguments RPAREN
+    | (IDENTIFIER|scopedId) LPAREN RPAREN
```

Comment:
That's fine, but what about making `arguments` optional like in:

```antlr
functionCall
    : (IDENTIFIER|scopedId) LPAREN arguments? RPAREN;
```

Can't tell if this makes a difference in performance or not, it's just a cosmetic change for clarity.

Reply:
Well it might be obsolete by now. I've seen this as a performance trick in some ANTLR grammars.